### ISSUE TITLE
[Python API] RuntimeError: Unknown parameter type <class 'uuid.UUID'> (Prepared statements)

### ISSUE THREAD

**Issue by sapalli2989** [kuzudb/kuzu#2923]
Thanks a lot for implementing native [UUID types](https://kuzudb.com/docusaurus/cypher/data-types/uuid)! It seems, Python API client does not support UUID in prepared statements yet?
```cypher
CREATE NODE TABLE V (id UUID, PRIMARY KEY(id));
```
```python
import uuid
conn.execute("CREATE (:V {id: $1});", {'1': uuid.uuid4()})
```

>Traceback (most recent call last):                                                          
  File "proj/kuzu_api.py", line 38, in execute
    results = conn.execute(query, parameters)  
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  
  File "proj/venv/lib64/python3.11/site-packages/kuzu/connection.py", line 82, in execute
    _query_result = self._connection.execute(  
                    ^^^^^^^^^^^^^^^^^^^^^^^^^  
RuntimeError: Unknown parameter type <class 'uuid.UUID'>

**Comment by andyfengHKU** [comment]
Hi @sapalli2989, 

thanks for reporting. I'll take a look at this issue. 

Meanwhile there is a workaround to pass uuid as a string. So something along the line of 

```
conn.execute("CREATE (:V {id: cast($1, "UUID")});", {'1': str(uuid.uuid4())})
```

**Comment by andyfengHKU** [comment]
@sapalli2989 FYI, this should be fixed with PR #2926

**Comment by sapalli2989** [comment]
@andyfengHKU Confirmed it works. Thanks!
